src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1159,6 +1159,11 @@ object Types {
     private[this] var lastSymbol: Symbol = _
     private[this] var checkedPeriod = Nowhere
 
+    /** Provided for debugging, can be used to intercept assignments of uncommon denotations
+     */  
+    private def checkDenot(d: Denotation)(implicit ctx: Context) = {
```

Comment:
We always update both `lastDenotation` and `lastSymbol`. And we already have a method that is a setter: `setDenot`, that has context and updates both. I would say that changing `computeDenot` to use this method and adding checks there feels cleaner.

This file already has 3k+ lines of code. And `NamedType` already has a checker, `checkSymAssign`. Introducing second one feels redundant.